shared/django_apps/executor.py

```diff
@@ -0,0 +1,17 @@
+from concurrent.futures import ThreadPoolExecutor
+
+_executor = None
+
+
+def get_executor():
+    global _executor
+    if _executor == None:
+        _executor = ThreadPoolExecutor(max_workers=1)
+    return _executor
+
+
+def run_in_executor(fn):
```

Comment:
[nit] suggest calling this `run_in_django_dedicated_thread` so it's clear why we're doing this